src/control/controlvalue.h

```diff
@@ -118,14 +113,14 @@ class ControlValueAtomicBase {
     // In worst case, each reader can consume a reader slot from a different ring element.
     // In this case there is still one ring element available for writing.
     ControlRingValue<T> m_ring[cRingSize];
-    std::atomic<int> m_readIndex;
-    std::atomic<int> m_writeIndex;
+    std::atomic<std::size_t> m_readIndex;
+    std::atomic<std::size_t> m_writeIndex;
```

Comment:
This doubles the size of the indexes in case of 64 bit archs and non atomic readable data types for no reason.
How about using `unsigned int`?  Even a `uint8_t` would be sufficient if, we are sure it `is_always_lock_free`
Please static_assert the later.

Reply:
yup, `uint8_t` also seems fine for me because I doubt there there would ever need to be a ring size that large. Using types smaller than native though (that includes `int` on x64_86) [do incur an extra `mov` instruction though](https://compiler-explorer.com/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:12,positionColumn:2,positionLineNumber:12,selectionStartColumn:2,selectionStartLineNumber:12,startColumn:2,startLineNumber:12),source:'%23include+%3Ccstdint%3E%0A%23include+%3Ccstddef%3E%0A%0Aextern+int+arr%5B512%5D%3B%0A%0Aint+getArr(int+i)+%7B%0A++++return+arr%5Bi%5D%3B%0A%7D%0A%0Aint+getArr(unsigned+int+i)+%7B%0A++++return+arr%5Bi%5D%3B%0A%7D%0A%0Aint+getArr(long+int+i)+%7B%0A++++return+arr%5Bi%5D%3B%0A%7D%0A%0Aint+getArr(std::size_t+i)+%7B%0A++++return+arr%5Bi%5D%3B%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-Wall+-Wconversion+-O3',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+14.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+14.2+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

Reply:
Interesting. OK, than let's keep std::size_t for the sake of the fastest solution.

Reply:
But please add an assertion for is_always_lock_free

Reply:
well, I've already made the change in a fixup, so we can keep it or drop it. I'm not to worried about a single `mov` instruction between registers (I haven't looked very deeply into it but I can't take more than a cycle). Shrinking the object might be a more significant speedup because it decreases cache pressure... I don't know. I'm personally not too worried about optimizing here because from looking at the rest of the codebase the eventually-consistent specialization is only used very rarely anyways... 
Two potentially interesting godbolts:
1. Due to padding, `std::size_t` vs `std::uint8_t` does barely make a difference. If it all, only for small types for which the ControlRingValue never gets instantiated. https://compiler-explorer.com/z/KcEEf3Ko7  
2. `is_always_lock_free` is also the case for types larger than `void*` on ARMv8 on clang: https://compiler-explorer.com/z/K57d1d5MP

In conclusion, we should keep `std::size_t` and (related to your other comment) rely on `is_always_lock_free`.